independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/producer/illegal/IllegalProducerTest.java

```diff
@@ -0,0 +1,59 @@
+package io.quarkus.arc.test.producer.illegal;
+
+import static org.junit.Assert.fail;
+
+import java.io.Serializable;
+import java.time.temporal.Temporal;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.inject.IllegalProductException;
+import javax.enterprise.inject.Produces;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.test.ArcTestContainer;
+
+public class IllegalProducerTest {
+
+    @Rule
+    public ArcTestContainer container = new ArcTestContainer(IllegalProducer.class);
+
+    @Test
+    public void testNormalScopedProducerMethodReturnsNull() {
+        Serializable serializable = Arc.container().instance(Serializable.class).get();
```

Comment:
@gastaldi Note that the invocation of `Instance.get()` does not result in `IllegalProductException` because the producers are called lazily. In other words, a proxy is returned and the producer is invoked when a proxy method is called -> a proxy is required to lookup the current bean instance per each invocation. This behavior is the same as in Weld.